tonic/models/vic/grid_params.py

```diff
@@ -661,64 +1218,172 @@ def grid_params(soil_dict, target_grid, snow_dict, veg_dict, veglib_dict,
                 for y, x in pyzip(ymask, xmask):
                     out_dict[var][:, :, y, x] = fill_val
 
+            elif mydict[var].ndim == 4:
+                j = mydict[var].shape[1]
+                k = mydict[var].shape[2]
+                l = mydict[var].shape[3]
+                out_dict[var] = np.ma.zeros((j, k, l, ysize, xsize),
+                                            dtype=dtype)
```

Comment:
The i, j, and k are used a couple lines later in the assignment of `out_dict` (e.g., line 1227: `for jj in pyrange(j):`  etc., then line 1230: `out_dict[var][jj, kk, ll, yi, xi] = mydict[var][:, jj, kk, ll]`).  Whoever had worked on grid_params.py as of spring 2016 (jhamman?) had implemented that logic. I assume the purpose was to map the mydict values, which only contain cells within the domain/basin, to out_dict, which additionally contains fill_values. I simply extended the logic to handle those state variables that have different dimensions than the variables found in the soil_param file. Can you suggest a better way of doing the assignment?

Personally I think that overhauling someone else's inefficient code should be part of a different PR. The purpose of my PR was simply to add needed functionality.  If the code prior to this PR was OK to merge into the archive, then my PR should be held to the same standard.